lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -541,29 +543,44 @@ func (c *ContainerProxy) Stop(vc *viccontainer.VicContainer, name string, second
 func (c *ContainerProxy) IsRunning(vc *viccontainer.VicContainer) (bool, error) {
 	defer trace.End(trace.Begin(""))
 
+	inspectJSON, err := c.dockerInfo(vc)
+	if err != nil {
+		return false, err
+	}
+	return inspectJSON.State.Running, nil
+}
+
+func (c *ContainerProxy) dockerInfo(vc *viccontainer.VicContainer) (*types.ContainerJSON, error) {
+	defer trace.End(trace.Begin(""))
+
 	if c.client == nil {
-		return false, InternalServerError("ContainerProxy.IsRunning failed to get a portlayer client")
+		return nil, InternalServerError("ContainerProxy.dockerInfo failed to get a portlayer client")
 	}
 
 	results, err := c.client.Containers.GetContainerInfo(containers.NewGetContainerInfoParamsWithContext(ctx).WithID(vc.ContainerID))
 	if err != nil {
 		switch err := err.(type) {
 		case *containers.GetContainerInfoNotFound:
-			return false, NotFoundError(fmt.Sprintf("No such container: %s", vc.ContainerID))
+			return nil, NotFoundError(fmt.Sprintf("No such container: %s", vc.ContainerID))
 		case *containers.GetContainerInfoInternalServerError:
-			return false, InternalServerError(err.Payload.Message)
+			return nil, InternalServerError(err.Payload.Message)
 		default:
-			return false, InternalServerError(fmt.Sprintf("Unknown error from the interaction port layer: %s", err))
+			return nil, InternalServerError(fmt.Sprintf("Unknown error from the interaction port layer: %s", err))
 		}
 	}
 
-	inspectJSON, err := ContainerInfoToDockerContainerInspect(vc, results.Payload, c.portlayerName)
+	return ContainerInfoToDockerContainerInspect(vc, results.Payload, c.portlayerName)
+}
+
+// IsBroken returns true if the given container is broken, e.g. migration failed
+func (c *ContainerProxy) IsBroken(vc *viccontainer.VicContainer) (bool, error) {
+	defer trace.End(trace.Begin(""))
+
+	inspectJSON, err := c.dockerInfo(vc)
 	if err != nil {
-		log.Errorf("containerInfoToDockerContainerInspect failed with %s", err)
 		return false, err
 	}
-
-	return inspectJSON.State.Running, nil
+	return inspectJSON.State.Status == "broken", nil
```

Comment:
Are there any statuses defined as constants? Using strings is not the best solution especially if it used more than 2 times across all project.
I also don't like IsBroken method. It's just awkward. There should be a method that returns status only it would be way more useful for other possible cases too.